### PR TITLE
AG-4088-Chat-Configure-Escalation-Period

### PR DESCRIPTION
**Pull request by sas-wtag**
[AG-4088](https://welltravel.atlassian.net/browse/AG-4088)

### CODE REVIEW

File: proto-libs/agent-platform/userRelationships.proto [hunk 1]
```diff
@@ -11,8 +11,14 @@ message UserRelationsCustomer {
...
+message ChatConfigurations {
+  int64 delayTime = 1;
+  bool sendNotification = 2;
+}
+
 service UserRelationships {
   rpc RetrieveCustomerRelations(Model.User) returns (UserRelationsCustomer) {}
   rpc RetrieveAgentCustomerRelations(Model.User) returns (stream Model.User) {}
   rpc RetrieveAgentColleagueRelations(Model.User) returns (stream Model.User) {}
+  rpc RetrieveChatConfigurations(Model.User) returns (ChatConfigurations) {}
```
[the remark this answers is not in the document]
Comment (sas-wtag) [reply proto-libs/agent-platform/userRelationships.proto]: added a new proto file

File: proto-libs/agent-platform/userPreferences.proto [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+package Agent;
+
+message ChatConfigurations {
+    int64 notificationEscalationDelayInSeconds = 1;
+    bool notificationEscalationEnabled = 2;
+}
+
+service UserPreferences {
+    rpc RetrieveChatConfigurations(Model.User) returns (ChatConfigurations) {}
+}
```
Comment (luxflux) [inline proto-libs/agent-platform/userPreferences.proto:14]:
```suggestion
}
```

File: proto-libs/agent-platform/userPreferences.proto [hunk 3]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+import "Model/user.proto";
+
+package Agent;
+
+message ChatConfigurations {
+    int64 notificationEscalationDelayInSeconds = 1;
```
Comment (luxflux) [inline proto-libs/agent-platform/userPreferences.proto:8]: Please check the indention, I think it should only be two spaces.

File: proto-libs/agent-platform/userPreferences.proto [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    bool notificationEscalationEnabled = 2;
+}
+
+service UserPreferences {
+    rpc RetrieveChatConfigurations(Model.User) returns (ChatConfigurations) {}
```
Comment (luxflux) [inline proto-libs/agent-platform/userPreferences.proto:13]:
```suggestion
    rpc RetrieveChatPreferences(Model.User) returns (ChatConfigurations) {}
```

File: proto-libs/agent-platform/userPreferences.proto [hunk 5]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+import "Model/user.proto";
+
+package Agent;
+
+message ChatConfigurations {
```
Comment (luxflux) [inline proto-libs/agent-platform/userPreferences.proto:7]: I'd call it preferences:
* A configuration is something more fundamental than a preference
* In the UI it's called preference as well - less confusion

```suggestion
message ChatPreferences {
```

**Review by luxflux — CHANGES_REQUESTED** [review]

File: proto-libs/agent-platform/userPreferences.proto (hunk 3 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userPreferences.proto]: done

File: proto-libs/agent-platform/userPreferences.proto (hunk 4 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userPreferences.proto]: done

File: proto-libs/agent-platform/userPreferences.proto (hunk 5 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userPreferences.proto]: done

**Review by luxflux — CHANGES_REQUESTED** [review]

File: index.js [hunk 6]
```diff
@@ -37,3 +38,7 @@ exports.Blabber.Agent.User = user.User;
 exports.Blabber.Email = require('./nodejs/agent-platform/email_grpc_pb');
 exports.Blabber.Email.SendEmailRequest = emailService.SendEmailRequest;
 exports.Blabber.Email.SendEmailResponse = emailService.SendEmailResponse;
+
+exports.Blabber.UserPreferences = require('./nodejs/agent-platform/userPreferences_grpc_pb');
+exports.Blabber.UserPreferences.ChatConfigurations =
+  userPreferencesService.ChatConfigurations;
```
[the remark this answers is not in the document]
Comment (sas-wtag) [reply index.js]: done

**Review by mal-wtag — APPROVED** [review]

**Pull request merged**